src/CoreWCF.Primitives/src/CoreWCF/Channels/Message.cs

```diff
@@ -830,6 +830,15 @@ internal void WriteMessagePostamble(XmlDictionaryWriter writer)
             }
         }
 
+        internal async Task WriteMessagePostambleAsync(XmlDictionaryWriter writer)
+        {
+            if (Version.Envelope != EnvelopeVersion.None)
+            {
+                await writer.WriteEndElementAsync();
```

Comment:
Not all writers support async IO. Not sure yet how to check if async is supported.